core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -742,7 +742,7 @@ private boolean unmanageNonRecursive(Entity e) {
             if (e instanceof Group) {
                 Collection<Entity> members = ((Group)e).getMembers();
                 for (Entity member : members) {
-                    if (!Entities.isNoLongerManaged(member)) member.removeGroup((Group)e);
+                    if (!Entities.isNoLongerManaged(member)) ((EntityInternal)member).groups().remove((Group)e);
```

Comment:
I guess what I am saying is that these are not the recommended calls, as far as I understand.  `.removeGroup()` is just doing the same thing under the hood as `groups().remove()`.  Should we not be using `Group#removeMember()`?